File: examples/javascript/handler-w-deps.js
// JavaScript example with in-memory dependency for GraalFaaS
// Uses a lightweight CommonJS-style require injected by the host.
const { greet } = require('greeter');

function handler(event) {
  const name = event && event.name != null ? event.name : "World";
  return { message: greet(name) };
}

File: examples/python/handler-w-deps.py
# Python example with an in-memory dependency for GraalFaaS
# The host preloads dependencies as modules (see manifest) so we can import them normally.
from greeter import greet

def handler(event):
    try:
        name = event.get("name") if isinstance(event, dict) else None
    except Exception:
        name = None
    return {"message": greet(name)}
